index.js

```diff
@@ -104,14 +103,45 @@ var postKinesisStream = function(records, exportFile, schemaData){
     Records: records_in_avro_format, /* required */
     StreamName: exportFile.postStream /* required */
   }
-  /*console.log("Not posting records")
-  console.log('Records to kinesis - ' + records_in_avro_format.length)*/
-  kinesis.putRecords(params, function (err, data) {
-    if (err) console.log(err, err.stack); // an error occurred
-    else    {
-      postedToKinesis += records_in_avro_format.length;
-      console.log("Successfully posted to kinesis: " + postedToKinesis);           // successful response
-    }
-  })
 
+  if(putRecordsUntilCompletelySuccessful (params, exportFile, true, records_in_avro_format)){
+    postedToKinesis += records_in_avro_format.length;
+    console.log("Successfully posted to kinesis: " + postedToKinesis); 
+  }
 }
+
+  var putRecordsUntilCompletelySuccessful = (params, exportFile, startRunning, recordsToPost) => {
```

Comment:
I think you can remove `recordsToPost` argument since it is always the same as `params.Records` (related comments below)